File: frontend/src/objects/ai/apiTypes/AIOrderProgressNarrativesResponse.ts
import type { OrderStatus } from '@/objects/shared/ids'

export interface AIOrderProgressNarrativeGroup {
  status: OrderStatus
  messages: string[]
}

export interface AIOrderProgressNarrativesResponse {
  groups: AIOrderProgressNarrativeGroup[]
  generatedAt: string
  generatedFor: string
}

File: backend/src/ai/objects/apiTypes/AIOrderProgressNarrativesResponse.scala
package delivery.ai.objects.apiTypes

import delivery.shared.objects.OrderStatus

final case class AIOrderProgressNarrativeGroup(
    status: OrderStatus,
    messages: List[String]
)

final case class AIOrderProgressNarrativesResponse(
    groups: List[AIOrderProgressNarrativeGroup],
    generatedAt: String,
    generatedFor: String
)
